File: Leetcode/JS/Leetcode495-提莫攻击.js
/**
 * @param {number[]} timeSeries
 * @param {number} duration
 * @return {number}
 */
 var findPoisonedDuration = function(timeSeries, duration) {
    let ans = 0,last = -1;
    for(const x of timeSeries){
        let t = x + duration - 1;
        ans += last < x ? duration : t - last;
        last = t;
    }
    return ans;
};

File: Leetcode/Rust/Leetcode495-提莫攻击.rs
impl Solution {
    pub fn find_poisoned_duration(time_series: Vec<i32>, duration: i32) -> i32 {
        let (mut ans,mut last) = (0,-1);
        for x in time_series.iter(){
            let t = x + duration - 1;
            if *x > last{
                ans += duration;
            }else{
                ans += t - last;
            }
            last = t;
        }
        ans
    }
}